ChessBoard.Api/Program.cs

```diff
@@ -0,0 +1,49 @@
+using ChessBoard.Api.Repositories;
+using ChessBoard.Api.Settings;
+using MongoDB.Driver;
+
+namespace ChessBoard.Api
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+            var services = builder.Services;
+
+            var serviceSettings = builder.Configuration
+                .GetSection(nameof(ServiceSettings))
+                .Get<ServiceSettings>();
+
+            services.AddSingleton(serviceProvider =>
```

Comment:
Maybe there are some better alternatives than singleton?
What do you think about scoped or transient?
Have a look into differences and consider some changes in this service registration part :)

Reply:
Hey I tried to dig deep to see if there would be differences between the lifetimes of the instance, and in this case there would not be much difference. I kind of understand where they differ, scoped - it uses same instance for any method that is called on it within the **same http request** while transient creates a new instance for every method invocation. But still I lack the experience to understand what is best to use for web api application. Any hints maybe? :)

Reply:
`Singleton`

A singleton service is created once and then used for the lifetime of the application. The same instance of the service is used wherever it is requested.

Use cases:

When the service needs to maintain some state for the lifetime of the application, you should use a singleton.
Services like logging, caching, or any other service which doesn't have any user-specific data and which is thread-safe can be registered as Singleton.
Singleton is also used when initializing objects that are expensive to create is needed, and that can be reused for the entire application life cycle.

`Transient`

A transient service is created each time it's requested. This means that each component or service that requires this service will get a new instance.

Use cases:

Transient lifetime services are light-weight, stateless services.
If a service doesn't maintain any state and does not have any issue being created multiple times, it should be registered as Transient.
Transient services are often used for stateless, lightweight services, like calculation services or validation services.

`Scoped`

A scoped service is created once per scope. In a web application, a scope is created for each request, so the service is created once per request.

Use cases:

When you need a service to maintain state within a single request or transaction, you should use a scoped service.
Services like database contexts or unit of work classes are often registered as Scoped as they need to maintain some state for the duration of a single request.
Scoped is also typically used when you need to share a single instance between different objects in the graph of objects handling a given request.
These lifetimes help to control how instances of services are created and shared, which can be critical in managing resources, performance, and preventing issues with concurrency in your applications.